### DISCUSSION TITLE
Will Casper 2.0 be a fork?

### DISCUSSION THREAD

**Discussion by sczembor** [casper-network/condor-info#23] [Q&A] [answered]
Will Casper 2.0 rollout happen as a usual upgrade? Or will it be a fork of the network?

**Comment by sczembor** [comment, marked as answer]
Casper 2.0 (Condor) is a major protocol upgrade that introduces new features and improvements. While it requires a transition from the previous version, it does not result in a separate, competing blockchain.

**Comment by sczembor** [comment]
@asladeofgreen

**Comment by sacherjj** [comment]
Technically, every upgrade of the Casper network is a fork. The upgrade mechanism is such that it allows synchronized upgrades of all nodes staged. So every upgrade since fast sync was released (1.5) had pieces that only allows nodes to talk with nodes running the same configuration.  

So at an upgrade point, we have 3 possible scenarios.

- 67% or more stake decide to not stage and upgrade and the majority of the Casper network continues on current network configuration through the upgrade point.  Those staged will hang until they remove the upgrade and restart.
- The weight of upgrade staged is between 33% and 66%, putting staged and unstaged weights with neither capable of moving the network forward.  This would require external governance to determine if we unstage or more stage to get one side to weight to finalize blocks.
- The way upgrades normal go is over 67% of weight stages. Those nodes that don't stage will pause at the last switch block of the pre-upgrade era. They need to stage the upgrade, put in a recent trusted hash and restart to continue down the upgraded path.

When the minority weight sides are hung, they are on a minority fork.

So an upgrade will be the same as normal upgrades as far as the nodes shutting down at the upgrade point and then running the new version and waiting for weight, then moving forward. The one big difference is that a sidecar is added to provide RPC. This will be installed as part of the staging process and it will wait until the node switched to Condor, then access the new binary port on the node and provide RPC over 7777 which the node had used with previous versions.

I'm starting to put together what this upgrade process looks like and we will also have info at a later time for this that want to run nodes attached to dev-net to test those setups and see how it differs from current node operation.
